### ISSUE TITLE
关于训练情况

### ISSUE THREAD

**Issue by bojone** [thu-coai/CDial-GPT#3]
论文说在数据集上训练了30个epoch，单步batch_size=8？我看了下语料有1200万组左右，按照单步0.1s算，总时间 1200万 * 30 / 8 * 0.1s 约等于52天？单卡训练了52天？？

**Comment by lemon234071** [comment]
> 论文说在数据集上训练了30个epoch，单步batch_size=8？我看了下语料有1200万组左右，按照单步0.1s算，总时间 1200万 * 30 / 8 * 0.1s 约等于52天？单卡训练了52天？？

苏神您好，论文我们使用了累积梯度64， 4卡训练，大概训练了20天左右 （论文昨天刚挂出来https://arxiv.org/abs/2008.03946）。“The
batch size was set to 8, and the number of gradient accumulation was set to
64. All the models trained with a vocabulary of 13,088 Chinese characters on
four RTX 2080ti GPUs.”。

**Comment by bojone** [comment]
噢噢，我就是看到了论文才发现这个项目的。做得相当不错！

那就是4卡训练了20天，等效batch_size是2048？我看你们训练了好几个版本的模型，都训练了20天？那不是加起来几个月了？

**Comment by lemon234071** [comment]
> 噢噢，我就是看到了论文才发现这个项目的。做得相当不错！
> 
> 那就是4卡训练了20天，等效batch_size是2048？我看你们训练了好几个版本的模型，都训练了20天？那不是加起来几个月了？

非常感谢您的关注! 
是的。
只有这个1200万数据的模型用了这么多天，那些小数据上的10天左右（其实具体多久记不清了，几个月前的），加上我占用了10多张卡，包括上调参试错，总计用了2个多月。

**Comment by bojone** [comment]
> > 噢噢，我就是看到了论文才发现这个项目的。做得相当不错！
> > 那就是4卡训练了20天，等效batch_size是2048？我看你们训练了好几个版本的模型，都训练了20天？那不是加起来几个月了？
> 
> 非常感谢您的关注!
> 是的。
> 只有这个1200万数据的模型用了这么多天，那些小数据上的10天左右（其实具体多久记不清了，几个月前的），加上我占用了10多张卡，包括上调参试错，总计用了2个多月。

虽然目前的参数设置应该也够用了，不过还是感觉没有用到相对位置编码有点遗憾，哈哈～

有时间的话，建议也转换个tf版出来，还是有一定需求的。

**Comment by lemon234071** [comment]
> > > 噢噢，我就是看到了论文才发现这个项目的。做得相当不错！
> > > 那就是4卡训练了20天，等效batch_size是2048？我看你们训练了好几个版本的模型，都训练了20天？那不是加起来几个月了？
> > 
> > 
> > 非常感谢您的关注!
> > 是的。
> > 只有这个1200万数据的模型用了这么多天，那些小数据上的10天左右（其实具体多久记不清了，几个月前的），加上我占用了10多张卡，包括上调参试错，总计用了2个多月。
> 
> 虽然目前的参数设置应该也够用了，不过还是感觉没有用到相对位置编码有点遗憾，哈哈～
> 
> 有时间的话，建议也转换个tf版出来，还是有一定需求的。

谢谢亲的建议，小店非常赞同这个建议，这边开始安排周转!

**Comment by bojone** [comment]
> > > > 噢噢，我就是看到了论文才发现这个项目的。做得相当不错！
> > > > 那就是4卡训练了20天，等效batch_size是2048？我看你们训练了好几个版本的模型，都训练了20天？那不是加起来几个月了？
> > > 
> > > 
> > > 非常感谢您的关注!
> > > 是的。
> > > 只有这个1200万数据的模型用了这么多天，那些小数据上的10天左右（其实具体多久记不清了，几个月前的），加上我占用了10多张卡，包括上调参试错，总计用了2个多月。
> > 
> > 
> > 虽然目前的参数设置应该也够用了，不过还是感觉没有用到相对位置编码有点遗憾，哈哈～
> > 有时间的话，建议也转换个tf版出来，还是有一定需求的。
> 
> 谢谢亲的建议，小店非常赞同这个建议，这边开始安排周转!

哈哈，等着，到时候集成到bert4keras中去～

**Comment by lemon234071** [comment]
> > > > > 噢噢，我就是看到了论文才发现这个项目的。做得相当不错！
> > > > > 那就是4卡训练了20天，等效batch_size是2048？我看你们训练了好几个版本的模型，都训练了20天？那不是加起来几个月了？
> > > > 
> > > > 
> > > > 非常感谢您的关注!
> > > > 是的。
> > > > 只有这个1200万数据的模型用了这么多天，那些小数据上的10天左右（其实具体多久记不清了，几个月前的），加上我占用了10多张卡，包括上调参试错，总计用了2个多月。
> > > 
> > > 
> > > 虽然目前的参数设置应该也够用了，不过还是感觉没有用到相对位置编码有点遗憾，哈哈～
> > > 有时间的话，建议也转换个tf版出来，还是有一定需求的。
> > 
> > 
> > 谢谢亲的建议，小店非常赞同这个建议，这边开始安排周转!
> 
> 哈哈，等着，到时候集成到bert4keras中去～

My pleasure!

**Comment by bojone** [comment]
冒昧问问，可以加qq或微信讨论下吗？有些细节想要进一步确认下。我的邮箱 bojone@spaces.ac.cn ，可以发qq或微信到邮箱，谢谢
